Runner/Runner.py

```diff
@@ -0,0 +1,120 @@
+import azure.common
+import logging
+import websockets
+import datetime
+import io
+import os
+import sys
+import time
+import json
+import string
+import asyncio
+import JobManager
+import Utils
+from os import listdir
+from os.path import isfile, join
+import azure.storage.blob as azureblob
+import azure.batch.batch_service_client as batch
+import azure.batch.batch_auth as batchauth
+import azure.batch.models as batchmodels
+import azext.batch as batch 
+import argparse
+
+sys.path.append('.')
+sys.path.append('..')
+
+_BATCH_ACCOUNT_NAME = os.environ['PS_BATCH_ACCOUNT_NAME']
+_BATCH_ACCOUNT_KEY = os.environ['PS_BATCH_ACCOUNT_KEY']
+_BATCH_ACCOUNT_URL = os.environ['PS_BATCH_ACCOUNT_URL']
+_BATCH_ACCOUNT_SUB = os.environ['PS_BATCH_ACCOUNT_SUB']
+_STORAGE_ACCOUNT_NAME = os.environ['PS_STORAGE_ACCOUNT_NAME']
+_STORAGE_ACCOUNT_KEY = os.environ['PS_STORAGE_ACCOUNT_KEY']
+_SERVICE_PRINCIPAL_CREDENTIALS_CLIENT_ID = os.environ['PS_SERVICE_PRINCIPAL_CREDENTIALS_CLIENT_ID']
+_SERVICE_PRINCIPAL_CREDENTIALS_SECRET = os.environ['PS_SERVICE_PRINCIPAL_CREDENTIALS_SECRET']
+_SERVICE_PRINCIPAL_CREDENTIALS_TENANT = os.environ['PS_SERVICE_PRINCIPAL_CREDENTIALS_TENANT']
+_SERVICE_PRINCIPAL_CREDENTIALS_RESOUCE = os.environ['PS_SERVICE_PRINCIPAL_CREDENTIALS_RESOUCE']
+
+_job_managers = []
+
+if __name__ == '__main__':
+
+    start_time = datetime.datetime.now().replace(microsecond=0)
+    print('Sample start: {}'.format(start_time))
+    print()
+
+    # Create the blob client, for use in obtaining references to
+    # blob storage containers and uploading files to containers.
+    
+    blob_client = azureblob.BlockBlobService(
+        account_name=_STORAGE_ACCOUNT_NAME,
+        account_key=_STORAGE_ACCOUNT_KEY)
+
+    credentials = ServicePrincipalCredentials(
+        client_id = _SERVICE_PRINCIPAL_CREDENTIALS_CLIENT_ID,
+        secret = _SERVICE_PRINCIPAL_CREDENTIALS_SECRET,
+        tenant = _SERVICE_PRINCIPAL_CREDENTIALS_TENANT,
+        resource = _SERVICE_PRINCIPAL_CREDENTIALS_RESOUCE)
+
+    batch_client = batch.BatchExtensionsClient(
+        credentials=credentials,
+        batch_account=_BATCH_ACCOUNT_NAME,
+        base_url=_BATCH_ACCOUNT_URL,
+        subscription_id=_BATCH_ACCOUNT_SUB)     
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("TestConfig", help="A manifest file that contains a list of all the jobs and pools you want to create.")
+    args = parser.parse_args()
+    
+    TestConfigurationFile = args.TestConfig #"Tests/TestConfiguration.json"
+
+    try:
+        with open(TestConfigurationFile) as f: 
+            template = json.load(f)
+        
+        #for i in range(0, len(template["tests"])):  # just testing out 2 files for testing. 
+        for i in range(0, 2):  
+            test = template["tests"][i]
+            applicationLicenses = None
+
+            try:
+                applicationLicenses = test["applicationLicense"]
+            except KeyError:
+                pass
```

Comment:
💡 just idea share: you can either return some sort of `invalid` status here or some exit mechanism, using `pass` will jsut silently run the whole on on except `keyError` in this instance. There are few other places with similar pattern.

Reply:
I can just drop the KeyError in the try/catch. I'm wanting to just continue on because that json element can be empty by design.